maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java

```diff
@@ -238,10 +380,11 @@ public int compareTo( Item item )
                     return -1; // 1.any < 1-1
 
                 default:
-                    throw new RuntimeException( "invalid item: " + item.getClass() );
+                    throw new IllegalStateException( "invalid item: " + item.getClass() );
```

Comment:
per Java spec, this should be a ClassCastException

Reply:
I'm not convinced:
- this is private implementation detail that should never happen, then if it happen, I find this IllegalStateException quite a good idea
- moreover, we're not doing a class cast, then this ClassCastException IMHO would be confusing

Reply:
OK, it's not really a Java type issue, I suppose.  Perhaps Item should be an enum and then this problem goes away.